testinfra/backend/kubectl.py

```diff
@@ -21,23 +21,41 @@ class KubectlBackend(base.BaseBackend):
     NAME = "kubectl"
 
     def __init__(self, name, *args, **kwargs):
-        self.name, self.user = self.parse_containerspec(name)
+        self.container = None
+        self.name, self.container = self.parse_podspec(name)
+
+        if kwargs:
+            if "namespace" in kwargs:
+                self.namespace = kwargs.get('namespace')
+            if "container" in kwargs:
+                self.container = kwargs.get('container')
```

Comment:
``parse_containerspec()`` also handled the user defined with ``user@pod``, but I see that ``kubectl`` doesn't allow to exec as arbitrary user, so I'm ok with your change.

If a user provide container in both way: ``pod/a?container=b`` we will use one of them but it's unclear to me, I'd prefer to raise or not parse one of the container (which will lead to raise in kubectl).

So I think ```__init__``` could be:

```python
self.namespace = kwargs.get('namespace')
self.container = kwargs.get('container')
if self.container is None:
    self.name, self.container = self.parse_podspec(name)
else:
   self.name = name
super(KubectlBackend, self).__init__(self.name, *args, **kwargs)
````